### PR TITLE
Fix rare replicas diverge

### PR DESCRIPTION
**Pull request by alesapin**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Bug Fix (user-visible misbehaviour in official stable or prestable release)

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix extremely rare case when ReplicatedMergeTree replicas can diverge after hard reboot of all replicas. The error looks like `Part  ... intersects (previous|next) part ...`

### CODE REVIEW

File: src/Storages/StorageReplicatedMergeTree.cpp [hunk 1]
```diff
@@ -4298,7 +4288,7 @@ void StorageReplicatedMergeTree::startup()
 
     try
     {
-        queue.initialize(getDataParts());
+        queue.initialize(getZooKeeper());
```
Comment (tavplubix) [inline src/Storages/StorageReplicatedMergeTree.cpp:4291]: Btw, should we move this call to `ReplicatedMergeTreeRestartingThread`?

**Review by tavplubix — APPROVED** [review]

**Comment by alesapin** [comment]
Sanitizer race already reported: https://github.com/ClickHouse/ClickHouse/issues/28854

**Comment by tavplubix** [comment]
Functional stateless tests (release, DatabaseReplicated) - `Part 8_149_193_2 intersects previous part 8_6_163_18`

**Comment by tavplubix** [comment]
Stress test (thread) - `Part all_4_93_1 intersects previous part all_0_67_1`. 
Too many intersecting parts, looks suspicious.

**Comment by alesapin** [comment]
And again intersecting parts. Probably fix is invalid.

**Comment by alesapin** [comment]
Ok, one more time 
@Mergifyio update

**Comment by alesapin** [comment]
One more time
@Mergifyio update

**Comment by alesapin** [comment]
I'll try for the last time 
@Mergifyio update

**Comment by alesapin** [comment]
got some intersecting parts

**Comment by alesapin** [comment]
One more time (only stress and functional is important)
@Mergifyio update

**Comment by alesapin** [comment]
Unrelated: `Child process was terminated by signal 9 (KILL). If it is not done by 'forcestop' command or manually, the possible cause is OOM Killer`

**Pull request merged**